andela/mervels-bn-backend
# Errors should be handled well and logged in a log file

#### What does this PR do?
- implements the functionality of logging the errors into a log file.
#### Description of Task to be completed?
- catching all errors
- logging them into a log file
#### How should this be manually tested?
- clone the repo
- create a directory called `logs` inside the `src folder` inside directory `logs` create a file called `barefoot.log`
- run npm install to install the dependencies `sentry and winston`
- include the environment variable of `SENTRY_DSN`
- start the server
- force an error
-you shall see an error logged on the console, in the barefoot.log file you created, and on sentry.io

#### Any background context you want to provide?
All methods with a try-catch in the controllers have been changed to accept an additional `next` parameter, and a `return next(error)` in catch block to direct the errors to the error handler
#### What are the relevant pivotal tracker stories?
[#168131426](https://www.pivotaltracker.com/story/show/168748728)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/39488016/65670108-056e7100-e045-11e9-8e7e-6926f9fa53ea.png)
![image](https://user-images.githubusercontent.com/39488016/65670159-17501400-e045-11e9-92db-e40ddbd3d37c.png)

#### Questions:
N/A
